apps/web/src/app/dashboard/[name]/settings/page.tsx

```diff
@@ -1,3 +1,240 @@
-export default function Settings() {
-  return <div className="flex flex-col gap-4">Settings</div>;
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { Trash2 } from "lucide-react";
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import {
+  type SubmitErrorHandler,
+  type SubmitHandler,
+  useForm,
+} from "react-hook-form";
+
+import AlertModal from "~/components/modals/alertModal";
+import { Button } from "~/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "~/components/ui/form";
+import { Input } from "~/components/ui/input";
+import { Skeleton } from "~/components/ui/skeleton";
+import { useToast } from "~/components/ui/use-toast";
+import {
+  type UpdateOrganizationName,
+  updateOrganizationNameSchema,
+} from "~/lib/validators/organization";
+import {
+  deleteOrganization,
+  getMembersOfOrganization,
+  getOwnOrganizationByName,
+  updateOrganizationName,
+} from "~/server/actions/organization";
+
+type SettingsMembersViewProps = {
+  params: {
+    name: string;
+  };
+};
+
+function SettingsGeneral({ params }: SettingsMembersViewProps) {
+  const { toast } = useToast();
+  const router = useRouter();
+  const session = useSession();
+  const [isAlertOpen, setIsAlertOpen] = useState(false);
+
+  const {
+    data: organization,
+    isLoading,
+    isFetched,
+  } = useQuery({
+    queryKey: ["organization", params.name],
+    queryFn: async () => {
+      const [organization, err] = await getOwnOrganizationByName(params.name);
+
+      console.log(organization, err);
+
+      if (err !== null) {
+        console.error(err);
+      }
+
+      if (organization == undefined) {
```

Comment:
Organization cannot equal to `undefined`. It can equal to `null`. Move this to the block where you're checking for an error.

Reply:
done